pkg/kubelet/kubelet.go

```diff
@@ -808,6 +808,18 @@ func NewMainKubelet(ctx context.Context,
 	klet.runner = runtime
 	klet.allocationManager.SetContainerRuntime(runtime)
 
+	// Register pod sandbox ready callback with runtime manager.
+	// The Type assertion is used because the `SetPodSandboxReadyCallback` method
```

Comment:
why should it not be part of the interface?

Reply:
Hello @haircommander,  do you mean adding the `OnPodSandboxReady` callback to [RuntimeHelper](https://github.com/kubernetes/kubernetes/blob/7d353c524988a1321ce63bc2e9bea90728b65105/pkg/kubelet/container/helpers.go#L49-L78) interface directly and and calling it as `m.runtimeHelper.OnPodSandboxReady()` in `containerruntime.SyncPod()`? 

I tried that approach and that was cleaner as well.
My only thought was if extending the `RuntimeHelper` interface was fine, provided all components implementing it will now need to define this additional function as well.

But now I think that is simpler approach, since within Kubernetes codebase, there're only 2 places which implements this interface  (Kubelet and FakeRuntimeHelper).

I'll update the PR.